v4/core/utils_test.go

```diff
@@ -192,3 +194,221 @@ func TestPointers(t *testing.T) {
+	expected := []string{"38", "26", "22"}
+	convertedSlice := ConvertSlice(int64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	int64Slice = []int64{}
+	convertedSlice = ConvertSlice(int64Slice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceInt(t *testing.T) {
+	intSlice := []int{3, 2, 1}
+	expected := []string{"3", "2", "1"}
+	convertedSlice := ConvertSlice(intSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	intSlice = []int{}
+	convertedSlice = ConvertSlice(intSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceByteArray(t *testing.T) {
+	testString := "test string 1..."
+	byteArray := []byte(testString)
+	byteArraySlice := [][]byte{byteArray}
+
+	// base64 encoded value
+	expected := []string{"dGVzdCBzdHJpbmcgMS4uLg=="}
+	convertedSlice := ConvertSlice(byteArraySlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	byteArraySlice = [][]byte{}
+	convertedSlice = ConvertSlice(byteArraySlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceDate(t *testing.T) {
+	date1 := strfmt.Date(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC))
+	date2 := strfmt.Date(time.Date(2020, time.November, 10, 23, 0, 0, 0, time.UTC))
+
+	dateSlice := []strfmt.Date{date1, date2}
+	expected := []string{"2009-11-10", "2020-11-10"}
+	convertedSlice := ConvertSlice(dateSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	dateSlice = []strfmt.Date{}
+	convertedSlice = ConvertSlice(dateSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceDateTime(t *testing.T) {
+	date1 := strfmt.DateTime(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC))
+	date2 := strfmt.DateTime(time.Date(2020, time.November, 10, 23, 0, 0, 0, time.UTC))
+
+	dateTimeSlice := []strfmt.DateTime{date1, date2}
+	expected := []string{"2009-11-10T23:00:00.000Z", "2020-11-10T23:00:00.000Z"}
+	convertedSlice := ConvertSlice(dateTimeSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	dateTimeSlice = []strfmt.DateTime{}
+	convertedSlice = ConvertSlice(dateTimeSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.Empty(t, convertedSlice)
+}
+
+func TestConvertSliceString(t *testing.T) {
+	stringSlice := []string{"testString1", "testString2"}
+	expected := []string{"testString1", "testString2"}
+	convertedSlice := ConvertSlice(stringSlice)
+
+	assert.NotNil(t, convertedSlice)
+	assert.NotEmpty(t, convertedSlice)
+	assert.Equal(t, expected, convertedSlice)
+
+	stringSlice = []string{"\"testString1\"", "\"testString2\""}
```

Comment:
I think if a string within the slice actually contains double quotes, then we want those double quotes to appear in the resulting string slice that is returned.